### DISCUSSION TITLE
Pickling Error when using MUSDB18 Dataset

### DISCUSSION THREAD

**Discussion by MCalafato** [asteroid-team/asteroid#658] [Q&A]
I am trying to train a TasNet model using the MUSDB18 dataset and have ran into some issues where I couldn't find fix for.

when adjusting the training script for the TasNet with the MUSDB18 dataloader and running the script, I am being given the following error: 
`_pickle.PicklingError: Can't pickle <function MUSDB18Dataset.<lambda> at 0x000001DBE2A3E310>: attribute lookup MUSDB18Dataset.<lambda> on asteroid.data.musdb18_dataset failed`

Does anyone have an idea of what might be the problem or how to fix it please?

**Comment by faroit** [comment]
@MCalafato I'be not seen that error yet. Can you send a reproducible script?

**Reply by MCalafato** [reply]
Yes sure. These are the 4 scripts I am using:

[scripts.zip](https://github.com/asteroid-team/asteroid/files/10928867/scripts.zip)

**Reply by MCalafato** [reply]
@faroit are these files enough to reproduce the error please?

**Reply by faroit** [reply]
thanks. will try this over the course of this week

**Reply by MCalafato** [reply]
Just found out that the error occurs when I use the num_workers larger than 0. However when I set it to 0, I get an error message saying CUDA out of memory.

`torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.69 GiB (GPU 0; 6.00 GiB total capacity; 3.89 GiB already allocated; 204.66 MiB free; 3.90 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF`

**Reply by MCalafato** [reply]
@faroit did you have any chance to look at it please? Thanks
